frame/examples/offchain-worker-price-oracle/src/lib.rs

```diff
@@ -0,0 +1,529 @@
+		NewPrice { price: u32, maybe_who: Option<T::AccountId> },
+		/// Event generated when a new authority is added.
+		AuthorityAdded { authority: T::AccountId },
+		/// Event generated when an authority is removed.
+		AuthorityRemoved { authority: T::AccountId },
+	}
+
+	/// A vector of recently submitted prices.
+	///
+	/// This is used to calculate average price, should have bounded size.
+	#[pallet::storage]
+	#[pallet::getter(fn prices)]
+	pub(super) type Prices<T: Config> = StorageValue<_, BoundedVec<u32, T::MaxPrices>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn authorities)]
+	pub(super) type Authorities<T: Config> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxAuthorities>, ValueQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NotAuthority,
+		AlreadyAuthority,
+		TooManyAuthorities,
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Offchain Worker entry point.
+		///
+		/// By implementing `fn offchain_worker` you declare a new offchain
+		/// worker. This function will be called when the node is fully synced
+		/// and a new best block is successfully imported.
+		///
+		/// Note that it's not guaranteed for offchain workers to run on EVERY
+		/// block, there might be cases where some blocks are skipped, or for
+		/// some the worker runs twice (re-orgs), so the code should be able to
+		/// handle that.
+		///
+		/// You can use `Local Storage` API to coordinate runs of the worker.
+		fn offchain_worker(block_number: BlockNumberFor<T>) {
+			// Note that having logs compiled to WASM may cause the size of the
+			// blob to increase significantly. You can use `RuntimeDebug` custom
+			// derive to hide details of the types in WASM. The `sp-api` crate
+			// also provides a feature `disable-logging` to disable all logging
+			// and thus, remove any logging from the WASM.
+			log::info!("Hello World from offchain workers!");
+
+			// Since off-chain workers are just part of the runtime code, they
+			// have direct access to the storage and other included pallets.
+			//
+			// We can easily import `frame_system` and retrieve a block hash of
+			// the parent block.
+			let parent_hash = <system::Pallet<T>>::block_hash(block_number - 1u32.into());
+			log::debug!("Current block: {:?} (parent hash: {:?})", block_number, parent_hash);
+
+			// It's a good practice to keep `fn offchain_worker()` function
+			// minimal, and move most of the code to separate `impl` block. Here
+			// we call a helper function to calculate current average price.
+			// This function reads storage entries of the current state.
+			let average: Option<u32> = Self::average_price();
+			log::debug!("Current price: {:?}", average);
+
+			/// A friendlier name for the error that is going to be returned in
+			/// case we are in the grace period.
+			const RECENTLY_SENT: () = ();
```

Comment:
```suggestion
```

I dont think this adds much.

Reply:
I do understand the author's original intent to have a descriptive error. So in the places where it is used, replace it with an empty tuple?
https://github.com/paritytech/substrate/blob/a08af900907f27bcad68479b59c7d4c8f400ea55/frame/examples/offchain-worker-price-oracle/src/lib.rs#L257-L258

Reply:
Also want to note that this came from the original implementation:
- https://github.com/paritytech/substrate/blob/a6b0d435995d7101e70ee0aab1707d0807024a3f/frame/examples/offchain-worker/src/lib.rs#L399

Please let me know how you'd like me to move forward.

Reply:
Please return `()` as error without any alias, it is not a good rust pattern.

Reply:
99ff6572f5bb0e3429d0fa028e3ff6c50a4050fe